### PR TITLE
DEVX-748: Augment Confluent Cloud Java example with Avro and Confluent Cloud Schema Registry

### PR DESCRIPTION
**Pull request by ybyzek**

### CODE REVIEW

**Review by joel-hamill — APPROVED** [review]

File: clients/cloud/java/README.md [hunk 1]
```diff
@@ -59,70 +60,142 @@ The Kafka Streams API reads the same topic from Confluent Cloud and does a state
...
+    ```shell
+    $ cat $HOME/.ccloud/config
+    ...
+    basic.auth.credentials.source=USER_INFO
+    schema.registry.basic.auth.user.info=<SR API KEY>:<SR API SECRET>
+    schema.registry.url=https://<SR ENDPOINT>
+    ...
+    ```
+
+4. Run the Avro producer, passing in arguments for (a) the local file with configuration parameters to connect to your Confluent Cloud instance and (b) the topic name:
```
Comment (NathanNam) [inline clients/cloud/java/README.md:159]: I think I missed something here. Was a topic called test2 created?

**Review by NathanNam — COMMENTED** [review]
I left one comment. Other than that, LGTM!

File: clients/cloud/java/README.md (hunk 1 above)
Comment (ybyzek) [reply clients/cloud/java/README.md:159]: @NathanNam good question.  I added a comment to the README that explains:

```
    # If the topic does not already exist, the code will use the Kafka Admin Client API to create the topic
```

Comment (NathanNam) [reply clients/cloud/java/README.md:159]: Thanks for adding this part.

File: clients/cloud/confluent-cli/README.md [hunk 2]
```diff
@@ -65,3 +65,59 @@ alice	{"count":2}
 When you are done, press `<ctrl>-c`.
 
 3. To demo the above commands, you may also run the provided script [confluent-cli-example.sh](confluent-cli-example.sh).
+
+
+
+
+# Example 2: With Fully-managed Confluent Schema Registry
+
+In this example, the producer writes Kafka data to a topic in Confluent Cloud, with fully-managed Confluent Schema Registry
```
Comment (NathanNam) [inline clients/cloud/confluent-cli/README.md]: @ybyzek I think we have not indicated that a user needs to "Enable Schema Registry" before she runs the following demo. Could we add a brief instruction on how to enable schema registry on Confluent Cloud. At the same time, we need to call out the VPC limitation we have.

```To use Confluent Cloud Schema Registry, your VPC must communicate with the Confluent Cloud Schema Registry public internet endpoint.```

**Review by NathanNam — APPROVED** [review]

File: clients/cloud/confluent-cli/README.md (hunk 2 above)
Comment (ybyzek) [reply clients/cloud/confluent-cli/README.md]: @NathanNam , it now reads:

```
...
Before using Confluent Cloud Schema Registry, check its availability and limits. Note that your VPC must be able to connect to the Confluent Cloud Schema Registry public internet endpoint.

1. As described in the Confluent Cloud quickstart, in the Confluent Cloud GUI, enable Confluent Cloud Schema Registry and create an API key and secret to connect to it.
...
```

**Comment by joel-hamill** [comment]
@ybyzek my related PR https://github.com/confluentinc/docs/pull/1631 is targeting `5.1.0-post`. any chance you can retarget yours?

File: clients/cloud/java/README.md [hunk 3]
```diff
@@ -59,70 +61,144 @@ The Kafka Streams API reads the same topic from Confluent Cloud and does a state
...
+1. As described in the [Confluent Cloud quickstart](https://docs.confluent.io/current/quickstart/cloud-quickstart.html), in the Confluent Cloud GUI, enable Confluent Cloud Schema Registry and create an API key and secret to connect to it.
+
+2. Verify your Confluent Cloud Schema Registry credentials work from your host. In the output below, substitute your values for `<SR API KEY>`, `<SR API SECRET>`, and `<SR ENDPOINT>`.
+
+    ```shell
+    # View the list of registered subjects
+    $ curl -u <SR API KEY>:<SR API SECRET> https://<SR ENDPOINT>/subjects
+    ```
+
+3. Add the following parameters to your local Confluent Cloud configuration file. In the output below, substitute values for `<SR API KEY>`, `<SR API SECRET>`, and `<SR ENDPOINT>`.
```
Comment (joel-hamill) [inline clients/cloud/java/README.md]: suggest adding a pointer to the config file location:

```
...local Confluent Cloud configuration file (``~/.ccloud/config``).
```

**Comment by ybyzek** [comment]
@joel-hamill I tried but rebasing isn't straightforward.  How important is to go back to 5.1.0-post?  For CCloud, why wouldn't latest CP (i.e. 5.1.2-post) be good enough?

**Comment by joel-hamill** [comment]
@ybyzek it could be good enough. i have = been adding all new cloud features to `5.1.0-post`, since it's the oldest version shown on the version picker, by default (this is doc team convention, for non-version specific content). alternatively i could rebase my SR changes `5.1.2-post`.

**Pull request merged**